File: sdk/java/src/main/java/vn/hitechcloud/tools/HiTechCloudToolsClient.java
package vn.hitechcloud.tools;

import vn.hitechcloud.tools.resources.*;
import java.io.IOException;
import java.util.Map;

public class HiTechCloudToolsClient {
    private final HttpClient http;
    public final MangAndHaTangResource mangAndHaTang;
    public final TenMienAndSslResource tenMienAndSsl;
    public final EmailAndDnsResource emailAndDns;
    public final CongCuLapTrinhResource congCuLapTrinh;
    public final SeoAndWebResource seoAndWeb;
    public final DuLieuVietNamResource duLieuVietNam;
    public final XuatNhapKhauAndLogisticsResource xuatNhapKhauAndLogistics;
    public final DuocPhamAndYTeResource duocPhamAndYTe;
    public final MaHoaAndKiemTraResource maHoaAndKiemTra;
    public final ChuyenDoiAndDinhDangResource chuyenDoiAndDinhDang;
    public final ThoiGianAndLichResource thoiGianAndLich;
    public final TaiChinhAndTyGiaResource taiChinhAndTyGia;
    public final PhapLyAndThuTucResource phapLyAndThuTuc;
    public final QrAndThanhToanResource qrAndThanhToan;
    public final DoanhNghiepAndThueResource doanhNghiepAndThue;

    public HiTechCloudToolsClient(String apiKey) {
        this(apiKey, "https://api-tools.hitechcloud.vn", 30000);
    }

    public HiTechCloudToolsClient(String apiKey, String baseUrl, int timeout) {
        this.http = new HttpClient(baseUrl, apiKey, timeout);
        this.mangAndHaTang = new MangAndHaTangResource(http);
        this.tenMienAndSsl = new TenMienAndSslResource(http);
        this.emailAndDns = new EmailAndDnsResource(http);
        this.congCuLapTrinh = new CongCuLapTrinhResource(http);
        this.seoAndWeb = new SeoAndWebResource(http);
        this.duLieuVietNam = new DuLieuVietNamResource(http);
        this.xuatNhapKhauAndLogistics = new XuatNhapKhauAndLogisticsResource(http);
        this.duocPhamAndYTe = new DuocPhamAndYTeResource(http);
        this.maHoaAndKiemTra = new MaHoaAndKiemTraResource(http);
        this.chuyenDoiAndDinhDang = new ChuyenDoiAndDinhDangResource(http);
        this.thoiGianAndLich = new ThoiGianAndLichResource(http);
        this.taiChinhAndTyGia = new TaiChinhAndTyGiaResource(http);
        this.phapLyAndThuTuc = new PhapLyAndThuTucResource(http);
        this.qrAndThanhToan = new QrAndThanhToanResource(http);
        this.doanhNghiepAndThue = new DoanhNghiepAndThueResource(http);
    }
}

File: sdk/csharp/HiTechCloud.Tools/HiTechCloudToolsClient.cs
namespace HiTechCloud.Tools;

using HiTechCloud.Tools.Resources;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text.Json;

public class HiTechCloudToolsClient
{
    private readonly HttpClient _httpClient;
    private readonly string _apiKey;

    public MangAndHaTangResource MangAndHaTang { get; }
    public TenMienAndSslResource TenMienAndSsl { get; }
    public EmailAndDnsResource EmailAndDns { get; }
    public CongCuLapTrinhResource CongCuLapTrinh { get; }
    public SeoAndWebResource SeoAndWeb { get; }
    public DuLieuVietNamResource DuLieuVietNam { get; }
    public XuatNhapKhauAndLogisticsResource XuatNhapKhauAndLogistics { get; }
    public DuocPhamAndYTeResource DuocPhamAndYTe { get; }
    public MaHoaAndKiemTraResource MaHoaAndKiemTra { get; }
    public ChuyenDoiAndDinhDangResource ChuyenDoiAndDinhDang { get; }
    public ThoiGianAndLichResource ThoiGianAndLich { get; }
    public TaiChinhAndTyGiaResource TaiChinhAndTyGia { get; }
    public PhapLyAndThuTucResource PhapLyAndThuTuc { get; }
    public QrAndThanhToanResource QrAndThanhToan { get; }
    public DoanhNghiepAndThueResource DoanhNghiepAndThue { get; }

    public HiTechCloudToolsClient(string apiKey, string baseUrl = "https://api-tools.hitechcloud.vn", int timeout = 30)
    {
        _apiKey = apiKey;
        _httpClient = new HttpClient { BaseAddress = new Uri(baseUrl), Timeout = TimeSpan.FromSeconds(timeout) };
        _httpClient.DefaultRequestHeaders.Add("X-API-Key", apiKey);

        MangAndHaTang = new MangAndHaTangResource(_httpClient);
        TenMienAndSsl = new TenMienAndSslResource(_httpClient);
        EmailAndDns = new EmailAndDnsResource(_httpClient);
        CongCuLapTrinh = new CongCuLapTrinhResource(_httpClient);
        SeoAndWeb = new SeoAndWebResource(_httpClient);
        DuLieuVietNam = new DuLieuVietNamResource(_httpClient);
        XuatNhapKhauAndLogistics = new XuatNhapKhauAndLogisticsResource(_httpClient);
        DuocPhamAndYTe = new DuocPhamAndYTeResource(_httpClient);
        MaHoaAndKiemTra = new MaHoaAndKiemTraResource(_httpClient);
        ChuyenDoiAndDinhDang = new ChuyenDoiAndDinhDangResource(_httpClient);
        ThoiGianAndLich = new ThoiGianAndLichResource(_httpClient);
        TaiChinhAndTyGia = new TaiChinhAndTyGiaResource(_httpClient);
        PhapLyAndThuTuc = new PhapLyAndThuTucResource(_httpClient);
        QrAndThanhToan = new QrAndThanhToanResource(_httpClient);
        DoanhNghiepAndThue = new DoanhNghiepAndThueResource(_httpClient);
    }
}
